gen/encoder.go

```diff
@@ -452,7 +452,11 @@ func (g *Generator) genStructMarshaler(t reflect.Type) error {
 	if !g.noStdMarshalers {
 		fmt.Fprintln(g.out, "// MarshalJSON supports json.Marshaler interface")
 		fmt.Fprintln(g.out, "func (v "+typ+") MarshalJSON() ([]byte, error) {")
-		fmt.Fprintln(g.out, "  w := jwriter.Writer{}")
+		if g.floatFmt != "" {
```

Comment:
Unnecessary if

Reply:
I'd kept an if condition to keep it compatible with older versions that don't have this struct field. I guess this can be removed if we are going to tag it as a version bump. What do you think?

Reply:
Of course, we will bump the version after merge

Reply:
I've removed the if condition.